ee/clickhouse/models/test/utils/util.py

```diff
@@ -0,0 +1,14 @@
+from time import sleep
+
+from ee.clickhouse.client import sync_execute
+
+
+# this normally is unnecessary as CH is fast to consume from Kafka when testing
+# but it helps prevent potential flakiness
+def delay_until_clickhouse_consumes_from_kafka(table_name: str, target_row_count: int) -> None:
+    # max = 10 seconds
```

Comment:
What do you think about something like:

```
ts_start = time.time()
timeout = 10

while time.time() < ts_start + timeout:
    result = sync_execute(f"SELECT COUNT(1) FROM {table_name}")
    if result[0][0] == target_row_count:
        return
    sleep(0.5)
```
?

Reply:
Agree this is clearer, updated!